### PR TITLE
Do not copy population on callback and add test

### PR DESCRIPTION
**Pull request by rogiervandergeer**
This is supposed to solve #132 .

In addition to this we can also ensure that fitness is copied on population copy. I'm inclined to do that as well, in a separate PR.

### CODE REVIEW

**Comment by koaning** [comment]
Mhmm... 

I wonder if something might go wrong. All of the other verbs are stateless but we now have a callback method that might change the population state. I'm wondering if this is more the responsibility of the end user or of the library.

**Comment by koaning** [comment]
Wouldn't this be a valid alternative?

```
callback_function(copy(self).evaluate(lazy=True), **kwargs)
```

Feel free to disagree with me by the way, I'm mostly curious about the considerations.

**Comment by rogiervandergeer** [comment]
Evaluating the full population in my current use case take a minute, that seems like a waste of time.

File: evol/population.py [hunk 1]
```diff
@@ -279,7 +279,7 @@ def callback(self, callback_function: Callable[['BasePopulation'], None],
         :return:
         """
         self.evaluate(lazy=True)
```
Comment (koaning) [inline evol/population.py]: but we're still evaluating the entire population here right? or is the copying the expensive part?

Comment (rogiervandergeer) [reply evol/population.py]: Those evaluations may already be present (hence the `lazy`), and can be used in the future.

**Pull request merged**